src/include/common/static_vector.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <cstring>
+#include <memory>
+#include <type_traits>
+#include <utility>
+
+#include "common/assert.h"
+
+namespace kuzu {
+namespace common {
+
+template<typename T>
+class MaybeUninit {
+    std::aligned_storage_t<sizeof(T), alignof(T)> data;
```

Comment:
Run into this https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1413r2.pdf, it seems `aligned_storage` will be deprecated. Any concerns over this?

Reply:
I read https://stackoverflow.com/questions/71828288/why-is-stdaligned-storage-to-be-deprecated-in-c23-and-what-to-use-instead and it makes me sad because they dont provide alternatives.

The only real alternative is super boring, it's to do all the manual work and create a properly aligned byte array.

Anyway. I guess I'll do the future-proof solution.